irmaclient/client.go

```diff
@@ -589,14 +589,24 @@ func (client *Client) credCandidates(request irma.SessionRequest, con irma.Attri
 			// then the entire conjunction is unsatisfiable
 			satisfiable = false
 		}
-		if len(c) == 0 {
-			if client.addCredSuggestion(request, credTypeID) {
-				// No acceptable credentials found. Excluding some nonsensical cases,
-				// add an "empty" credential (i.e. without hash) as a suggestion to the user
-				c = append(c, &credCandidate{Type: credTypeID})
+		// Determine whether we force an attribute value for any attribute requested from this credential here.
+		haveFixedAttrValue := false
+		for _, attr := range con {
+			if attr.Type.CredentialTypeIdentifier() != credTypeID {
+				continue
+			}
+			if attr.Value != nil {
+				haveFixedAttrValue = true
```

Comment:
The `have` here refers to something in the session request, while below the `have` in `haveCandidates` refers to something in our attribute wallet. Upon first glance, this variable name suggests to me that (when true) we have the required value in our wallet, which is not what this means. Suggestion: just `fixedAttrValue`.

Reply:
fixed it, good suggestion